okapi/services/caches/save_user_coords/WebService.php

```diff
@@ -0,0 +1,145 @@
+                'cache_code' => $cache_code,
+                'fields' => 'internal_id'
+            ))
+        );
+        $cache_id = $geocache['internal_id'];
+
+        self::update_coordinates($cache_id, $request->token->user_id, $latitude, $longitude);
+
+        $result = array(
+            'success' => true
+        );
+        return Okapi::formatted_response($request, $result);
+    }
+
+    private static function update_coordinates($cache_id, $user_id, $latitude, $longitude)
+    {
+        if (Settings::get('OC_BRANCH') == 'oc.de')
+        {
+
+            /* See:
+             *
+             * - https://github.com/OpencachingDeutschland/oc-server3/tree/development/htdocs/src/Oc/Libse/CacheNote
+             * - https://www.opencaching.de/okapi/devel/dbstruct
+             */
+
+            $rs = Db::query("
+                select max(id) as id
+                from coordinates
+                where
+                    type = 2  -- personal note
+                    and cache_id = '".Db::escape_string($cache_id)."'
+                    and user_id = '".Db::escape_string($user_id)."'
+            ");
+            $id = null;
+            if($row = Db::fetch_assoc($rs)) {
+                $id = $row['id'];
+            }
+            if ($id == null) {
+                Db::query("
+                    insert into coordinates (
+                        type, latitude, longitude, cache_id, user_id, description
+                    ) values (
+                        2,
+                        '".Db::escape_string($latitude)."',
+                        '".Db::escape_string($longitude)."',
+                        '".Db::escape_string($cache_id)."',
+                        '".Db::escape_string($user_id)."',
+                        '".Db::escape_string("")."'
+                    )
+                ");
+            } else {
+                Db::query("
+                    update coordinates
+                    set latitude  = '".Db::escape_string($latitude)."',
+                        longitude = '".Db::escape_string($longitude)."'
+                    where
+                        id = '".Db::escape_string($id)."'
+                        and type = 2
+                ");
+            }
+        }
+        else # oc.pl branch
+        {
+            $rs = Db::query("
+                select max(id) as id
+                from cache_mod_cords
+                where
+                    cache_id    = '".Db::escape_string($cache_id)."'
+                    and user_id = '".Db::escape_string($user_id)."'
+            ");
+            $id = null;
+            if($row = Db::fetch_assoc($rs)) {
+                $id = $row['id'];
+            }
+            if ($id == null) {
+                Db::query("
+                    insert into cache_mod_cords (
+                        cache_id, user_id, latitude, longitude
+                    ) values (
+                        2,
```

Comment:
Again, I see that you have a different number of values and args here. You'd really need to test this on OCPL first, before thinking of merging.